x-pack/plugins/actions/server/routes/create.ts

```diff
@@ -46,10 +45,17 @@ export const createActionRoute = (router: IRouter, licenseState: LicenseState) =
       }
       const actionsClient = context.actions.getActionsClient();
       const action = req.body;
-      const actionRes: ActionResult = await actionsClient.create({ action });
-      return res.ok({
-        body: actionRes,
-      });
+      try {
+        const actionRes: ActionResult = await actionsClient.create({ action });
+        return res.ok({
+          body: actionRes,
+        });
+      } catch (e) {
+        if (e instanceof ActionTypeDisabledError) {
+          return e.sendResponse(res);
+        }
```

Comment:
In the context of our conversation about handling the Api Keys error differently, perhaps we could do this more generically using a duck-typing approach.

Have a Type like:
```ts
interface ErrorThatHandlesItsOwnResponse {
  sendResponse(res: KibanaResponse);
}

function isErrorThatHandlesItsOwnResponse(e: Error) : e is ErrorThatHandlesItsOwnResponse {
  return typeof (e as ErrorThatHandlesItsOwnResponse)['sendResponse'] === 'function';
}
```

and then in our Routes we could do: 

```
if (isErrorThatHandlesItsOwnResponse(e)) {
  return e.sendResponse(res);
}
```

Reply:
Ya, I immediately thought of some Boom-like approaches here.  Suggest we use the "remove Boom" issue https://github.com/elastic/kibana/issues/49822 as the place to look into something like this.  Wanna add a comment there if you agree, @gmmorris ?

Reply:
@pmuellr @gmmorris let me know if what I added looks good: https://github.com/elastic/kibana/pull/59070/commits/b3f1282f5e1eda6ae2023e262384d6f244c1fbd5

Reply:
Yeah, that looks good!

One little thing we _could_ do is wrap all the handlers in a single error handling function that calls the handler in a try catch and that way the special casing would only appear in one place:
```
try {
 ... call handler
      } catch (e) {
        if (isErrorThatHandlesItsOwnResponse(e)) {
          return e.sendResponse(res);
        }
        throw e;
      }
```

Like the Boom errors hadling.

But it's no biggie :)